packages/portfolio-contract/src/portfolio.contract.ts

```diff
@@ -10,20 +12,34 @@ import type { ZCF } from '@agoric/zoe';
 import type { ResolvedPublicTopic } from '@agoric/zoe/src/contractSupport/topics.js';
 import type { Zone } from '@agoric/zone';
 import type { CopyRecord } from '@endo/pass-style';
-import { M } from '@endo/patterns';
+import { M, mustMatch } from '@endo/patterns';
 import { preparePortfolioKit } from './portfolio.exo.ts';
 import * as flows from './portfolio.flows.ts';
-import { makeProposalShapes } from './type-guards.ts';
+import {
+  makeProposalShapes,
+  makeOfferArgsShapes,
+  type OfferArgsShapes,
+} from './type-guards.ts';
 
 const trace = makeTracer('PortC');
 
 const interfaceTODO = undefined;
 
+const privateArgsShape = {
+  ...(OrchestrationPowersShape as CopyRecord),
+  marshaller: M.remotable('marshaller'),
+  contract: M.splitRecord({
+    aavePool: M.string(),
+    compound: M.string(),
+    factory: M.string(),
+  }),
+  chainInfo: M.recordOf(M.string(), ChainInfoShape),
+  assetInfo: M.arrayOf([M.string(), DenomDetailShape]),
+  poolMetricsNode: M.remotable(),
```

Comment:
`poolMetricsNode` is a fast usdc thing; it shouldn't be here.

I think I fixed that in my deploy package pr.

We can keep it here for now, but add a TODO